### PR TITLE
Arm Code

### PR DESCRIPTION
**Pull request by El1zabethS**

### CODE REVIEW

File: chargedUp/src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -91,18 +77,47 @@ public void drivetrainDefaultCommand(){
...
+        onTrue(new InstantCommand(()-> m_armSubsystem.spinEndEffector(0.5)))
+        .onFalse(new InstantCommand(()-> m_armSubsystem.stopEndEffector()));
+        
+        
+        // This runs Endeffector to eject game peices
+        m_auxController.rightTrigger().
+        onTrue(new InstantCommand(()-> m_armSubsystem.spinEndEffector(-0.2)))
+        .onFalse(new InstantCommand(()-> m_armSubsystem.stopEndEffector()));
+  
+        m_auxController.y().
```
Comment (J-Palomino) [inline chargedUp/src/main/java/frc/robot/RobotContainer.java:104]: Can we add a comment saying what this is intended to do

File: chargedUp/src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -91,18 +77,47 @@ public void drivetrainDefaultCommand(){
...
+        onTrue(new RotationPID(m_armSubsystem, 77.5));
+        
+        m_auxController.x(). onTrue(new RotationPID(m_armSubsystem, -77.5));
```
Comment (J-Palomino) [inline chargedUp/src/main/java/frc/robot/RobotContainer.java:107]: needs description

File: chargedUp/src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -91,18 +77,47 @@ public void drivetrainDefaultCommand(){
...
+        m_auxController.leftBumper().onTrue(new InstantCommand(()-> m_armSubsystem.resetRotationPosition()));
+  }
+  
   /**
    * Use this to pass the autonomous command to the main {@link Robot} class.
    *
    * @return the command to run in autonomous
    */
+
```
Comment (J-Palomino) [inline chargedUp/src/main/java/frc/robot/RobotContainer.java:116]: this function isnt doing anything do we need it?

File: chargedUp/src/main/java/frc/robot/Robot.java [hunk 4]
```diff
@@ -51,7 +51,9 @@ public void robotPeriodic() {
 
   /** This function is called once each time the robot enters Disabled mode. */
   @Override
-  public void disabledInit() {}
+  public void disabledInit() {
+    m_robotContainer.disablePIDSubsystems();
```
Comment (J-Palomino) [inline chargedUp/src/main/java/frc/robot/Robot.java:55]: This disabled a function that doesn't have any functional code in it

File: chargedUp/src/main/java/frc/robot/RobotContainer.java [hunk 5]
```diff
@@ -91,18 +77,47 @@ public void drivetrainDefaultCommand(){
    * joysticks}.
    */
   private void configureBindings() {
-    //Spins Motor if April Tags are Recognized for 20 Ticks
-    m_driverController.a().
-        onTrue(new targetFinding(m_drivetrainSubsystem, m_visionSubsystem));
+        m_auxController.start().
```
Comment (J-Palomino) [inline chargedUp/src/main/java/frc/robot/RobotContainer.java:80]: please add comment

File: chargedUp/src/main/java/frc/robot/commands/ArmControls/TelescoperReset.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  public void end(boolean interrupted) {
+    m_armSubsystem.spinTelescopingMotor(0);
+    m_armSubsystem.resetTelescoperPosition();
+    m_armSubsystem.telescoperCurrentLimit(30, 60);
+  }
+
+  // Returns true when the command should end.
+  @Override
+  public boolean isFinished() {
+    return m_armSubsystem.telecoperCurrent() >= 20;
```
Comment (J-Palomino) [inline chargedUp/src/main/java/frc/robot/commands/ArmControls/TelescoperReset.java:41]: have we tested this value? and will it vary with battery voltage?

File: chargedUp/src/main/java/frc/robot/subsystems/ArmSubsystem.java [hunk 7]
```diff
@@ -0,0 +1,136 @@
+// Copyright (c) FIRST and other WPILib contributors.
```
Comment (J-Palomino) [inline chargedUp/src/main/java/frc/robot/subsystems/ArmSubsystem.java:1]: This file is cracked really good foundation for the arm subsystem

File: chargedUp/src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java [hunk 8]
```diff
@@ -73,7 +73,7 @@ public DrivetrainSubsystem() {
     rightFollower1 = new SparkMaxWrapper(DriveMotor.rightFollower1.CAN_ID, MotorType.kBrushless);
 
     Arrays.asList(leftLead, leftFollower1, rightLead, rightFollower1)
-        .forEach((CANSparkMax spark) -> spark.setIdleMode(IdleMode.kBrake));
+        .forEach((CANSparkMax spark) -> spark.setIdleMode(IdleMode.kCoast));
```
Comment (J-Palomino) [inline chargedUp/src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java:76]: Awesome use of forEach! 👍

**Review by J-Palomino — CHANGES_REQUESTED** [review]
good job on getting this in an organized overall still a bit of a large pr but we will get there. The arm subsystem looks awesome! With all those methods you should be able to get it to do whatever you want! all this needs is a few more comments in robot container and then decide if you want to leave that disablePIDSubsystems method or pull it outr until it does something

File: chargedUp/src/main/java/frc/robot/Robot.java (hunk 4 above)
Comment (El1zabethS) [reply chargedUp/src/main/java/frc/robot/Robot.java:55]: removed function 👍

File: chargedUp/src/main/java/frc/robot/RobotContainer.java (hunk 1 above)
Comment (El1zabethS) [reply chargedUp/src/main/java/frc/robot/RobotContainer.java:104]: 👍

File: chargedUp/src/main/java/frc/robot/commands/ArmControls/TelescoperReset.java (hunk 6 above)
Comment (El1zabethS) [reply chargedUp/src/main/java/frc/robot/commands/ArmControls/TelescoperReset.java:41]: this isn't tested lol deleted file (code is stored in different branch)

File: chargedUp/src/main/java/frc/robot/subsystems/ArmSubsystem.java (hunk 7 above)
Comment (El1zabethS) [reply chargedUp/src/main/java/frc/robot/subsystems/ArmSubsystem.java:1]: :D

File: chargedUp/src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java (hunk 8 above)
Comment (El1zabethS) [reply chargedUp/src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java:76]: :D

**Review by El1zabethS — COMMENTED** [review]
we fixed it :)

**Comment by J-Palomino** [comment]
If this is ready to go with you please merge @El1zabethS

**Pull request merged**